### PR TITLE
[Feat] Add minipc chassis example

### PR DESCRIPTION
**Pull request by jerrywrx**

### CODE REVIEW

File: examples/chassis/typeC.cc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  while (true) {
+    chassis->SetSpeed(dbus->ch0, dbus->ch1, dbus->ch2);
+
+   // Kill switch
+    if (dbus->swr == remote::UP || dbus->swr == remote::DOWN) {
+     RM_ASSERT_TRUE(false, "Operation killed");
+    }
+
+    chassis->Update(false, 30, 20, 60);
```
Comment (ywh114) [inline examples/chassis/typeC.cc:73]: nit: Consider naming arguments, or comment usage? Same for SetSpeed above.
// Update(bool power_limit_on, float power_limit, float chassis_power, float chassis_power_buffer)

File: examples/minipc/ChassisTest.cc [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                                             .tz_module = 0,
+                                             .reserved = 0};
+osThreadId_t chassisTaskHandle;
+
+void chassisTask(void* argument) {
+  UNUSED(argument);
+  uint32_t flags;
+  control::MotorCANBase* motors[] = {fl_motor, fr_motor, bl_motor, br_motor};
+
+  float vx_keyboard = 0, vy_keyboard = 0, wz_keyboard = 0;
```
Comment (ywh114) [inline examples/minipc/ChassisTest.cc:88]: nit: Why different treatment for vX and vX_keyboard? Consider using newlines.

File: examples/minipc/ChassisTest.cc [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+  while (true) {
+    uint32_t RGB_color[3] = {0xFFFF0000, 0xFF00FF00, 0xFF0000FF};
```
Comment (ywh114) [inline examples/minipc/ChassisTest.cc:91]: What is the use case for this? Cycling through colors? If so, I think a dedicated successor function on enums would make more sense.
I grepped the entire repo for this and RGB/rgb_color only shows up here. Correct me if I am wrong.

File: examples/minipc/ChassisTest.cc [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+    // Keyboard control / Navigation mode
+    if (dbus->swr == remote::UP) {
+      flags = 0;
```
Comment (ywh114) [inline examples/minipc/ChassisTest.cc:95]: Can this be removed?

File: examples/minipc/ChassisTest.cc [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        fr_motor->SetOutput(0);
+        bl_motor->SetOutput(0);
+        br_motor->SetOutput(0);
+
+        led->Display(RGB_color[0]);
+      }
+      control::MotorCANBase::TransmitOutput(motors, 4);
+      osDelay(10);
+    }
+
```
Comment (ywh114) [inline examples/minipc/ChassisTest.cc:123]: Remove newline.

File: examples/minipc/ChassisTest.cc [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    // Dbus control
+    else {
```
Comment (ywh114) [inline examples/minipc/ChassisTest.cc:125]: } else {  // Dbus control

File: examples/minipc/ChassisTest.cc [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+      vy_keyboard = vy;
+      wz_keyboard = vw;
+
+      // When timeout it returns -2 so we need extra checks here
+      if (flags != osFlagsErrorTimeout && flags & DATA_READY_SIGNAL) {
+        vx_keyboard = clip<float>(vx_keyboard, -1200, 1200);
+        vy_keyboard = clip<float>(vy_keyboard, -1200, 1200);
+
+        chassis->SetSpeed(vx_keyboard, vy_keyboard, wz_keyboard);
+        chassis->Update(false, 30, 20, 60);
```
Comment (ywh114) [inline examples/minipc/ChassisTest.cc:108]: nit: same as `examples/chassis/typeC.cc`. Also below on line 126/7.

File: examples/minipc/ChassisTest.cc [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+      flags = osEventFlagsWait(chassis_flag_id, DATA_READY_SIGNAL, osFlagsWaitAny, 50);
+
+      vx_keyboard = vx;
+      vy_keyboard = vy;
+      wz_keyboard = vw;
+
+      // When timeout it returns -2 so we need extra checks here
+      if (flags != osFlagsErrorTimeout && flags & DATA_READY_SIGNAL) {
+        vx_keyboard = clip<float>(vx_keyboard, -1200, 1200);
+        vy_keyboard = clip<float>(vy_keyboard, -1200, 1200);
```
Comment (ywh114) [inline examples/minipc/ChassisTest.cc:105]:
```
      // When timeout it returns -2 so we need extra checks here
      if (flags != osFlagsErrorTimeout && flags & DATA_READY_SIGNAL) {
        vx_keyboard = clip<float>(vx, -1200, 1200);
        vy_keyboard = clip<float>(vy, -1200, 1200);
        wz_keyboard = vw;
```

**Review by ywh114 — COMMENTED** [review]
Notes on minor style/decision choices.

File: examples/minipc/ChassisTest.cc (hunk 2 above)
Comment (jerrywrx) [reply examples/minipc/ChassisTest.cc:88]: Fixed.

File: examples/minipc/ChassisTest.cc (hunk 3 above)
Comment (jerrywrx) [reply examples/minipc/ChassisTest.cc:91]: Fixed. Was unaware of available color variables.

File: examples/minipc/ChassisTest.cc (hunk 4 above)
Comment (jerrywrx) [reply examples/minipc/ChassisTest.cc:95]: Fixed.

File: examples/minipc/ChassisTest.cc (hunk 6 above)
Comment (jerrywrx) [reply examples/minipc/ChassisTest.cc:125]: Fixed.

File: examples/minipc/ChassisTest.cc (hunk 8 above)
Comment (jerrywrx) [reply examples/minipc/ChassisTest.cc:105]: Fixed.

File: examples/minipc/ChassisTest.cc [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+        // if timeout (no packet, stop the motor)
+        fl_motor->SetOutput(0);
+        fr_motor->SetOutput(0);
+        bl_motor->SetOutput(0);
+        br_motor->SetOutput(0);
+
+        led->Display(display::color_red);
+      }
+      control::MotorCANBase::TransmitOutput(motors, 4);
+      osDelay(10);
```
Comment (ywh114) [inline examples/minipc/ChassisTest.cc:118]: Also, this and line 122/123 can be joined together.

Comment (ywh114) [reply examples/minipc/ChassisTest.cc:118]: As a side effect helps to make `led->Display` more consistent :)

File: examples/minipc/ChassisTest.cc (hunk 4 above)
Comment (jerrywrx) [reply examples/minipc/ChassisTest.cc:95]: Which part should be removed?

Comment (ywh114) [reply examples/minipc/ChassisTest.cc:95]: `flags = 0;` seems extraneous.

Comment (jerrywrx) [reply examples/minipc/ChassisTest.cc]: Removed.

File: examples/minipc/ChassisTest.cc (hunk 9 above)
Comment (jerrywrx) [reply examples/minipc/ChassisTest.cc]: Fixed.

**Review by rickxu2 — APPROVED** [review]
LGTM

**Pull request merged**